offer-manager-domain/src/main/groovy/life/qbic/business/offers/identifier/OfferId.groovy

```diff
@@ -1,26 +1,70 @@
 package life.qbic.business.offers.identifier
 
+import groovy.transform.CompileStatic
+import groovy.transform.EqualsAndHashCode
+
 /**
- * Describes the offer identifier
+ * Represents and identifier for an offer
  *
- * Contains the random part, the project conserved part and the version from which the offer identifier is build
+ * Contains the random part, project conserved part and the version of an identifier
  *
  * @since 0.1.0
  */
-class OfferId extends TomatoId{
+@CompileStatic
+@EqualsAndHashCode(includeFields = true)
+class OfferId implements Comparable<OfferId>{
```

Comment:
you are changing the class hierarchy here, and therfore introduce a breaking change for clients. Please preserve the inheritance and add the interface to ensure backwards compatibility.

Reply:
As discussed this was necessary to simplify the code and will not affect older versions (as they do not use it)